File: assignment 3/c/programs/1.c
#include <stdio.h>

int main()
{
    int m, n;
    printf("Enter the number of rows: ");
    scanf("%d", &m);
    printf("Enter the number of columns: ");
    scanf("%d", &n);

    int matrix[m][n];

    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < n; j++)
        {
            printf("Enter element [%d][%d]: ", i, j);
            scanf("%d", &matrix[i][j]);
        }
    }

    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < n; j++)
        {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }

    int s = 0;
    for (int i = 0; i < m; i++)
    {
        s += matrix[i][i];
    }
    printf("sum of principal diagonal: %d\n", s);

    s = 0;
    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < n; j++)
        {
            if (i + j == m - 1)
            {
                s += matrix[i][j];
            }
        }
    }
    printf("sum of anti diagonal: %d\n", s);

    return 0;
}

File: assignment 3/python/programs/1.py
m = int(input("Enter the number of rows: "))
n = int(input("Enter the number of columns: "))

matrix = []

for i in range(m):
    row = []
    for j in range(n):
        element = int(input(f"Enter element [{i}][{j}]: "))
        row.append(element)
    matrix.append(row)

for i in range(m):
    for j in range(n):
        print(matrix[i][j], end=" ")
    print()

s = 0
for i in range(m):
    s = s + matrix[i][i]
print("sum of principle diagonal:", s)
s = 0
for i in range(m):
    for j in range(n):
        if i + j == m - 1:
            s = s + matrix[i][j]
print("sum of anti diagonal:", s)
